src/avram/database_validations.cr

```diff
@@ -21,11 +24,11 @@ module Avram::DatabaseValidations(T)
   # pass and Avram tries to save the record.
   private def validate_uniqueness_of(
     attribute : Avram::Attribute,
-    query : Avram::Criteria,
+    query : Avram::Criteria(T::BaseQuery, _),
```

Comment:
I had added generics to the Attribute and the Criteria to connect them. I think it might have worked but with my recent work on enums, I'm not sure if the types of the two would be the same so I took it off to be safe.